lms/djangoapps/certificates/views/support.py

```diff
@@ -77,27 +83,44 @@ def search_by_user(request):
         ]
 
     """
-    query = request.GET.get("query")
-    if not query:
-        return JsonResponse([])
+    user_query = request.GET.get("user_query", "")
+    if not user_query:
+        msg = _("user is not given.")
+        return HttpResponseBadRequest(msg)
 
     try:
-        user = User.objects.get(Q(email=query) | Q(username=query))
+        user = User.objects.get(Q(email=user_query) | Q(username=user_query))
     except User.DoesNotExist:
-        return JsonResponse([])
+        return HttpResponseBadRequest(_("user '{user}' does not exist").format(user=user_query))
 
     certificates = api.get_certificates_for_user(user.username)
     for cert in certificates:
         cert["course_key"] = unicode(cert["course_key"])
         cert["created"] = cert["created"].isoformat()
         cert["modified"] = cert["modified"].isoformat()
+        cert["regenerate"] = True
+
+    course_id = request.GET.get("course_id", "")
+    if course_id:
+        try:
+            course_key = CourseKey.from_string(course_id)
+        except InvalidKeyError:
+            return HttpResponseBadRequest(_("Course id '{course_id}' is not valid").format(course_id=course_id))
+        else:
+            if modulestore().get_course(course_key):
```

Comment:
Now that the CourseOverview model is available, we can use it instead of direct calls to modulestore for 'lightweight' existence checks and field inspections for courses.  CourseOverview will call modulestore if needed.